File: go/matrix.go
package main

func matrix(matrix1, matrix2 [][]int) [][]int {
	var result [][]int = make([][]int, matrixLen)

	for col := range matrix1 {
		result[col] = make([]int, matrixLen)
		for row := range result[col] {
			result[col][row] = matrix1[col][row] * matrix2[col][row]
		}
	}

	return result
}

File: java/src/main/java/Matrix.java
public class Matrix {
    public static int[][] Run(int[][] matrix1, int[][] matrix2) {
        int[][] result = new int[matrix1.length][matrix1.length];

        for (int row = 0; row < matrix1.length; row++) {
            for (int col = 0; col < matrix1[row].length; col++) {
                result[row][col] = matrix1[row][col] * matrix2[row][col];
            }
        }

        return result;
    }
}
